src/components/tickets/index.tsx

```diff
@@ -0,0 +1,159 @@
+
+    apiUrl += selectedPriorities.length
+      ? selectedPriorities.map((priority) => `&priority=${priority}`).join('')
+      : '';
+
+    return apiUrl;
+  }, [_sort, _order, selectedPriorities, paginationModel]);
+
+  const fetchTickets = useCallback(async () => {
+    try {
+      const apiUrl = buildApiUrl();
+      console.log(apiUrl);
+      const response = await TicketService.getAll(apiUrl);
+      setTotalRows(parseInt(response.headers['x-total-count']));
+      dispatch(setTicketState(response.data));
+    } catch (error) {
+      throw new Error('Failed to fetch tickets');
+    }
+  }, [buildApiUrl, dispatch, selectedPriorities, setSearchParams]);
+
+  const toggleModal = useCallback(() => {
+    setIsModalOpen((prev) => !prev);
+  }, []);
+
+  const toggleUpdateModal = useCallback(
+    (id?: string) => {
+      setSelectedTicket(storeTickets?.find((ticket) => ticket.id === id) || null);
+      setIsUpdateModalOpen((prev) => !prev);
+    },
+    [storeTickets],
+  );
+
+  const handleDeleteClick = useCallback(
+    async (rowId: string) => {
+      const deletedTicket = await TicketService.delete(rowId);
+      dispatch(deleteTicket(deletedTicket.data));
+      fetchTickets();
+    },
+    [dispatch, fetchTickets],
+  );
+
+  return (
+    <StyledBox>
+      {storeTickets.length > 0 && (
+        <StyledDataGrid
+          autoHeight
+          rows={storeTickets}
+          rowHeight={92}
+          columns={[
+            ...columns,
+            {
+              field: 'menu',
+              headerName: '',
+              width: 15,
+              sortable: false,
+              renderCell: (params) => {
+                return (
+                  <MenuCell
+                    id={params.row.id}
+                    toggleUpdateModal={toggleUpdateModal}
+                    handleDeleteClick={handleDeleteClick}
+                  />
+                );
+              },
+            },
+          ]}
+          paginationMode='server'
+          rowCount={totalRows}
+          pageSizeOptions={pageSizeOptions}
+          paginationModel={paginationModel}
+          onPaginationModelChange={setPaginationModel}
+          disableRowSelectionOnClick
+          disableColumnMenu
+          slots={{
+            toolbar: () => (
+              <CustomToolbar
+                dispatch={dispatch}
+                fetchTickets={fetchTickets}
+                toggleModal={toggleModal}
+                setSearchParams={setSearchParams}
+                selectedPriorities={selectedPriorities}
+                setSelectedPriorities={setSelectedPriorities}
+                isModalOpen={isModalOpen}
+              />
+            ),
+          }}
+        />
+      )}
+      {isUpdateModalOpen && (
+        <>
+          <TicketsModal
+            dispatch={dispatch}
+            toggleModal={toggleUpdateModal}
+            refetchTickets={fetchTickets}
+            initialValues={selectedTicket}
+            isEdit={true}
```

Comment:
Is it should be always `true`?

Reply:
if isEdit is set to true than Edit function is used if isEdit is not set or false the Create function in for Form submit is used.